tests/10apidoc/03events-initial.pl

```diff
@@ -126,18 +129,27 @@ sub flush_events_for
    });
 }
 
+# Note that semantics are undefined if calls are interleaved with differing
+# $room_ids for the same user.
 sub await_event_for
 {
-   my ( $user, $filter ) = @_;
+   my ( $user, %params ) = @_;
+
+   my $filter = $params{filter} || sub { 1 };
```

Comment:
FYI you could use `delete` here, as `delete` is much like Python's `dict.pop()` in that it returns the removed element.

    my $filter = delete $params{filter} || sub { 1 };

is a fairly common pattern